android/src/main/java/org/devio/rn/splashscreen/SplashScreen.java

```diff
@@ -14,21 +14,27 @@
  * Email:crazycodeboy@gmail.com
  */
 public class SplashScreen {
+    private static int NULL_ID = 0;
     private static Dialog mSplashDialog;
     private static WeakReference<Activity> mActivity;
 
     /**
      * 打开启动屏
      */
-    public static void show(final Activity activity, final boolean fullScreen) {
+    public static void show(final Activity activity, final boolean fullScreen, final int themeResId) {
         if (activity == null) return;
         mActivity = new WeakReference<Activity>(activity);
         activity.runOnUiThread(new Runnable() {
             @Override
             public void run() {
                 if (!activity.isFinishing()) {
 
-                    mSplashDialog = new Dialog(activity, fullScreen ? R.style.SplashScreen_Fullscreen : R.style.SplashScreen_SplashTheme);
+                    mSplashDialog = new Dialog(
+                            activity,
+                            themeResId != NULL_ID ? themeResId
```

Comment:
it would be best to check if it is `fullScreen` first.

Reply:
Actually, now that I'm thinking about it, there might be an even cleaner implementation here:


```java
// Not concerned with fullscreen anymore
public static void show(final Activity activity, final int themeResId) {
  ...
}

public static void show(Activity activity, boolean fullScreen) {
    int resourceId = fullScreen ? R.style.SplashScreen_Fullscreen : R.style.SplashScreen_SplashTheme;

    show(activity, resourceId);
}

```